discovery-provider/integration_tests/challenges/test_referral_challenge.py

```diff
@@ -151,3 +165,102 @@ def test_referral_challenge(app):
             .all()
         )
         assert len(challenges) == 10
+
+        # Test:
+        # Ensure there are no verified user referrals created yet
+
+        challenges = (
+            session.query(UserChallenge)
+            .filter(
+                UserChallenge.user_id == referrer.user_id,
+                UserChallenge.challenge_id == "referrals-verified",
+                UserChallenge.is_complete == True,
+            )
+            .all()
+        )
+        assert len(challenges) == 0
+
+        # Test: verified users
+        # - Ensure that verified user referrals aren't counted
+        #   for referrer credit
+        # - Ensure that a verified user challenge exists
+
+        verified_user = User(
+            blockhash="0x1",
+            blocknumber=BLOCK_NUMBER,
+            txhash="xyz",
+            user_id=12,
+            is_current=True,
+            handle="VerifiedReferrer",
+            handle_lc="verifiedreferrer",
+            wallet="0x1",
+            is_creator=False,
+            is_verified=True,
+            name="referrer_name",
+            created_at=datetime.now(),
+            updated_at=datetime.now(),
+        )
+        session.add(verified_user)
+        session.flush()
+
+        dispatch_new_user_signup(verified_user.user_id, 13, session, bus)
+        bus.flush()
+        bus.process_events(session)
+
+        # Ensure no regular referral created
+        challenges = (
+            session.query(UserChallenge)
+            .filter(
+                UserChallenge.user_id == verified_user.user_id,
+                UserChallenge.challenge_id == "referrals",
+                UserChallenge.is_complete == True,
+            )
+            .all()
+        )
+        assert len(challenges) == 0
+
+        # Ensure one verified referral created
+        challenges = (
+            session.query(UserChallenge)
+            .filter(
+                UserChallenge.user_id == verified_user.user_id,
+                UserChallenge.challenge_id == "referrals-verified",
+                UserChallenge.is_complete == True,
+            )
+            .all()
+        )
+        assert len(challenges) == 1
+
+        # Test: verified max count
+        #  - Ensure with > 500 verified referrals, we cap at 500
+        #  - No regular referrals are made
+
+        for i in range(510):
+            dispatch_new_user_signup(verified_user.user_id, 14 + i, session, bus)
```

Comment:
no-op needed, but the 14 + i is pretty brittle

Reply:
indeed